### PR TITLE
libzim 8.1.0 (new formula)

### PR DESCRIPTION
**Pull request by adamlamar**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by adamlamar** [comment]
`brew audit --new libzim` suggests using `uses_from_macos "icu4c"`, but this doesn't work:

```
Run-time dependency icu-i18n found: NO (tried pkgconfig and framework)

../meson.build:75:4: ERROR: Dependency "icu-i18n" not found, tried pkgconfig and framework
```

**Comment by adamlamar** [comment]
@chenrui333 would you be able to advise on the right approach for fixing the audit failure? When I set `uses_from_macos "icu4c"` the build for libzim doesn't work, failing with the `ERROR: Dependency "icu-i18n" not found` message mentioned above. I believe this is because the icu shipped with macos doesn't include all icu tooling.

Searching the github issues I didn't see other projects having this problem, but I did see [v8 brings in its own icu](https://github.com/chrmoritz/homebrew-core/blob/9c81126a576881c361c0dacd928ac996907d08e5/Formula/v8.rb#L39-L42). Is this the right approach?

File: Formula/libzim.rb
Comment (SMillerDev) [inline Formula/libzim.rb:39]:
```suggestion
    assert_match "libzim #{version}", shell_output("./test")
```

File: Formula/libzim.rb
Comment (SMillerDev) [inline Formula/libzim.rb:36]: Doesn't this work without `-std`?

File: Formula/libzim.rb
Comment (SMillerDev) [inline Formula/libzim.rb:24]: Why not
```suggestion
    system "meson", "setup", "build", *std_meson_args
    system "meson", "compile", "-C", "build", "--verbose"
    system "meson", "install", "-C", "build"
```

File: Formula/libzim.rb
Comment (adamlamar) [reply Formula/libzim.rb:36]: No, the header parsing fails without `-std=c+11`

File: Formula/libzim.rb
Comment (adamlamar) [reply Formula/libzim.rb:24]: I have no opinion - just following the homebrew meson template and the libzim documentation. This seems to work though. Is there is an advantage to using meson directly?

Comment (chenrui333) [reply Formula/libzim.rb:24]: updating the template, https://github.com/Homebrew/brew/pull/14275

**Review by chenrui333 — DISMISSED** [review]

**Review by carlocab — APPROVED** [review]

**Comment by kelson42** [comment]
I wonder why the PR is approved but the CI fails. I guess that means the CI has some kind of problem or misconfiguration?

File: Formula/libzim.rb
Comment (SMillerDev) [reply Formula/libzim.rb]: One less build dependency as far as I know. And we don't have to create a directory to work from. I'll update the template later.

**Comment by SMillerDev** [comment]
> I wonder why the PR is approved but the CI fails. I guess that means the CI has some kind of problem or misconfiguration?

Because our rules say you should use icu4c from macOS wherever you can. But CI isn't smart enough to determine when that is. Maintainers are so they approved the use of `depends_on "icu4c"` here, over CI.

**Comment by SMillerDev** [comment]
We should wait for the new icu4c update before re-running and merging this.

**Comment by kelson42** [comment]
> We should wait for the new icu4c update before re-running and merging this.

@SMillerDev Any ETA or ticket we could link here?

**Comment by SMillerDev** [comment]
We never give ETAs, it's done when it's done. The PR is https://github.com/Homebrew/homebrew-core/pull/113512 though

**Comment by kelson42** [comment]
@SMillerDev Your last comment there is encouraging :) Thank you very much for your help!

**Comment by adamlamar** [comment]
Should not be stale. Still waiting on https://github.com/Homebrew/homebrew-core/pull/113512

**Comment by chenrui333** [comment]
> Should not be stale. Still waiting on #113512

that PR got merged, so I think we can rebase and re-try this PR for the merge.

**Review by carlocab — DISMISSED** [review]

**Comment by kelson42** [comment]
Who can merge this?

**Review by carlocab — APPROVED** [review]
Rebased to get a fresh CI run. We can merge this when that's done.

**Comment by kelson42** [comment]
@adamlamar It seems there is still a small detail to fix to get the CI running. Could you please have a look?

**Pull request closed without merging**
